### PR TITLE
chore: test file types

### PR DESCRIPTION
**Pull request by samgermain**

### CODE REVIEW

File: ts/src/test/Exchange/base/test.sharedMethods.ts [hunk 1]
```diff
@@ -366,14 +368,14 @@ function removeProxyOptions (exchange: Exchange, skippedProperties: string[]) {
     return [ proxyUrl, httpProxy, httpsProxy, socksProxy ];
 }
 
-function setProxyOptions (exchange: Exchange, skippedProperties: string[], proxyUrl, httpProxy, httpsProxy, socksProxy) {
+function setProxyOptions (exchange: Exchange, skippedProperties: string[], proxyUrl: string, httpProxy: string, httpsProxy: string, socksProxy: string) {
     exchange.proxyUrl = proxyUrl;
     exchange.httpProxy = httpProxy;
     exchange.httpsProxy = httpsProxy;
     exchange.socksProxy = socksProxy;
 }
```
Comment (carlosmiei) [inline ts/src/test/Exchange/base/test.sharedMethods.ts:377]: @samgermain as @ttodua noted in a different PR skippedProperties shouldn't be typed like string[]

Comment (samgermain) [reply ts/src/test/Exchange/base/test.sharedMethods.ts:377]: > @samgermain as @ttodua noted in a different PR skippedProperties shouldn't be typed like string[]

do you have a link?
are you referring to `skippedProperties` specifically?

**Comment by samgermain** [comment]
@carlosmiei I don't understand why the appveyor has a failed build, when I look at it I can't see any problems

**Comment by samgermain** [comment]
unrelated build failure

```
FAILED bitrue (PHP Async ): (explain @ run-tests.js:353)

	

	[INFO] TESTING  php {"exchange":"bitrue","symbol":"all","isWs":false} 

	

	[INFO:MAIN] Exchange loaded 1810 symbols

	[INFO:MAIN] Selected SPOT SYMBOL: BTC/USDT

	[INFO:MAIN] Selected SWAP SYMBOL: FET/USDT:USDT

	[TEST_FAILURE] [ccxt\BadSymbol] bitrue does not have $market $symbol OMNIUSDT

	/home/travis/build/ccxt/ccxt/php/async/Exchange.php:3756

	/home/travis/build/ccxt/ccxt/php/async/bitrue.php:1600 ::: ccxt\async\Exchange > market
```

**Comment by samgermain** [comment]
[AppVeyor is failing, but no error listed in build](https://ci.appveyor.com/project/ccxt/ccxt/builds/49647595)

File: ts/src/pro/test/Exchange/test.watchOHLCV.ts [hunk 2]
```diff
@@ -21,7 +21,7 @@ async function testWatchOHLCV (exchange, skippedProperties, symbol) {
         let response = undefined;
         try {
             response = await exchange.watchOHLCV (symbol, chosenTimeframeKey, since, limit);
-        } catch (e) {
+        } catch (e: any) {
```
Comment (carlosmiei) [inline ts/src/pro/test/Exchange/test.watchOHLCV.ts:24]: why do we need this cast?

Comment (samgermain) [reply ts/src/pro/test/Exchange/test.watchOHLCV.ts]: Apparently we don't, but I feel like I was receiving an error before without the cast

Comment (samgermain) [reply ts/src/pro/test/Exchange/test.watchOHLCV.ts]: I think this is why https://ci.appveyor.com/project/ccxt/ccxt/builds/49701606

**Comment by samgermain** [comment]
unrelated failed build

```
FAILED kucoinfutures (JavaScript WS): (explain @ run-tests.js:358)

	

	[INFO] TESTING  js {

	  exchange: 'kucoinfutures',

	  symbol: undefined,

	  method: undefined,

	  isWs: true

	} 

	

	[INFO:MAIN] Exchange loaded 295 symbols

	[INFO:MAIN] Selected SWAP SYMBOL: BTC/USDT:USDT

	[TEST_FAILURE] [AssertionError] AssertionError [ERR_ASSERTION]: bids[0][0] (62917.2) should be < than asks[0][0] (62917.2) <<< kucoinfutures watchOrderBookForSymbols ::: {"bids":[[62925,30815],[62920.9,60],[62920.8,15],[62920.7,15],[62920.2,38],[62920.1,38],[62916.1,27],[62916,38],[62915.5,58],[62912.2,470],[62909.6,26],[62909.4,31],[62907.5,283],[62907.4,476],[62907.1,123],[62906.2,470],[62906.1,5],[62904.7,10],[62903.1,288],[62903,470],[62902.6,4],[62902.3,476],[62901.7,1],[62901.3,431],[62901.2,714],[62900.4,28],[62900.3,12],[62899.9,19],[62899.8,470],[62899.5,4],[62899.3,5],[62899.2,4],[62899,2],[62898.9,198],[62898.5,2],[62897.5,2],[62897.4,2],[62897.2,32],[62897.1,2],[62897,2],[62896.9,698],[62896.8,816],[62896.7,132],[62896.5,64],[62896.4,1105],[62896.3,32],[62896.2,32],[62896.1,2],[62896,32],[62895.1,2],[62894.9,3],[62894.5,7],[62894.4,473],[62893.7,286],[62893.6,32],[62893.5,5],[62893.3,417],[62893.2,475],[62893.1,476],[62892.6,7],[62892.2,432],[62891.8,714],[62891,56],[62890.9,15],[62890.7,210],[62890.5,2],[62890,28],[62889.9,32],[62889.7,34],[62889.4,290],[62889.3,32],[62888.8,32],[62888.6,32],[62888.2,7],[62888.1,15],[62887.3,33],[62887.2,724],[62887,64],[62886.6,388],[62885.9,2],[62885.8,10],[62885.2,26],[62885.1,32],[62885,32],[62884.8,2],[62884.7,131],[62884.4,470],[62883.6,7],[62882.4,2],[62882,16],[62881.6,20],[62880.8,490],[62880.5,314],[62879.8,2],[62879.6,4],[62879,2],[62878.8,415],[62878.7,692],[62878.5,31],[62878.1,2],[62877.8,505],[62877.2,290],[62877.1,198],[62877,3],[62876.7,471],[62876.6,2],[62876.4,4],[62876.3,32],[62876.1,4],[62876,2],[62875.2,32],[62875,32],[62874.9,5],[62874.7,286],[62874.4,2],[62873.9,346],[62871.7,21],[62870.9,5],[62870.7,9],[62870.5,1015],[62869.9,46],[62868.3,372],[62868,2400],[62867.9,3992],[62865.6,3],[62864.4,315],[62863,288],[62862.8,471],[62861.9,3],[62860.7,292],[62860.6,473],[62860,40],[62859.8,15],[62859.7,37],[62859.4,15],[62858.7,322],[62857.3,4265],[62857.2,37],[62855.7,5],[62855.6,3],[62855.4,288],[62855,188],[62853.9,32],[62852.8,62],[62852.6,32],[62852.3,126],[62850.2,2],[62849.4,595],[62849,71],[62848.9,17],[62843.8,77],[62843.1,1858],[62842.5,8],[62838.7,150],[62837.7,3],[62834,470],[62832.1,671],[62830,94],[62829.4,200],[62825.1,4],[62824.9,473],[62824.6,32],[62824.5,8],[62824.3,5],[62814.8,4706],[62812.5,5],[62811.7,5],[62810.3,55],[62808.9,55],[62806.5,796],[62805.4,5],[62798.6,314],[62795.8,15],[62780.4,3],[62741.6,796],[62641.8,618],[62278.7,7]],"asks":[[62917.2,54],[62925.1,11675],[62925.2,38],[62925.3,58],[62925.4,47],[62925.5,977],[62927.7,4],[62928.1,2],[62928.3,4],[62928.7,71],[62928.8,421],[62928.9,691],[62929.1,61],[62929.2,7],[62929.3,3],[62930,7],[62930.4,38],[62931.2,32],[62931.3,1013],[62931.4,158],[62931.5,290],[62931.7,20],[62931.8,34],[62931.9,2],[62932.3,32],[62932.4,13],[62933,68],[62933.6,3],[62933.8,8],[62934.4,34],[62934.5,615],[62934.7,470],[62935,5],[62935.2,4],[62935.3,1021],[62935.8,1689],[62936.2,1421],[62937.7,15],[62938.1,2],[62938.2,4],[62938.4,4],[62939.7,704],[62940,10],[62940.8,2],[62940.9,14],[62941.3,956],[62942,32],[62942.3,314],[62943.7,32],[62943.8,635],[62944,352],[62944.2,210],[62944.4,26],[62944.6,40],[62945.3,32],[62946.1,3992],[62946.9,17],[62948.6,1187],[62949.5,22],[62950.3,30],[62950.7,2],[62953.3,635],[62953.6,5],[62953.7,3],[62953.8,962],[62954.8,1589],[62955.9,1],[62957,73],[62957.7,47],[62958.8,37],[62959.6,282],[62960.3,32],[62961.3,38],[62961.9,1],[62962.6,3],[62963.3,472],[62966.7,8],[62967.7,32],[62968.8,3],[62969.6,2],[62970.5,315],[62971.5,3],[62975.9,2],[62979.6,1401],[62980.7,150],[62981.7,229],[62981.8,327],[62982.2,2],[62982.4,1],[62982.9,1],[62988.5,2],[62990,69],[62992.3,13],[62994.8,2],[62998.2,395],[63001,188],[63001.1,2],[63003.2,192],[63005.3,470],[63007.4,2],[63010,942],[63012.4,657],[63013.6,2],[63019.9,2],[63022.2,10],[63026.2,2],[63032.5,2],[63032.6,73],[63033.5,73],[63038.8,2],[63045.1,2],[63050.9,7533],[63054,2383],[63056.1,20],[63149.5,470],[63152.7,470],[63175.1,470],[63208.3,603],[63472.3,787],[63554.9,787],[63695.8,784]],"timestamp":1714199599308,"datetime":"2024-04-27T06:33:19.308Z","nonce":1701605486867,"symbol":"BTC/USDT:USDT"} >>> 

	    at testOrderBook (file:///home/travis/build/ccxt/ccxt/js/src/test/Exchange/base/test.orderBook.js:77:13)

	    at Object.testWatchOrderBookForSymbols [as watchOrderBookForSymbols] (file:///home/travis/build/ccxt/ccxt/js/src/pro/test/Exchange/test.watchOrderBookForSymbols.js:32:9)

	    at runNextTicks (node:internal/process/task_queues:60:5)

	    at listOnTimeout (node:internal/timers:538:9)

	    at process.processTimers (node:internal/timers:512:7)

	    at async callMethod (file:///home/travis/build/ccxt/ccxt/js/src/test/test.js:120:12)

	    at async testMainClass.testMethod (file:///home/travis/build/ccxt/ccxt/js/src/test/test.js:411:9)

	    at async testMainClass.testSafe (file:///home/travis/build/ccxt/ccxt/js/src/test/test.js:481:17)

	    at async Promise.all (index 6)

	    at async testMainClass.runTests (file:///home/travis/build/ccxt/ccxt/js/src/test/test.js:621:25) kucoinfutures (WS) swap watchOrderBookForSymbols [["BTC/USDT:USDT"]]

	[TEST_FAILURE] kucoinfutures (WS) swap PUBLIC_TESTS Failed methods : watchOrderBookForSymbols (explain @ run-tests.js:359)

FAIL ["kucoinfutures"] (run-tests.js:444)
```

**Pull request merged**